brite2caom2/storage_name.py

```diff
@@ -0,0 +1,181 @@
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  : 4 $
+#
+# ***********************************************************************
+#
+
+from os.path import basename, dirname, exists, splitext
+from caom2pipe.manage_composable import StorageName
+
+
+__all__ = ['add_entry', 'BriteName']
+
+
+class BriteName(StorageName):
+    """
+    For the decorrelated planes, need to have the content of multiple files to generate the previews, so implement
+    this class to collect the artifacts by plane product ID. This decision is captured in how _source_names and
+    _destination_uris are set.
+    """
+
+    def __init__(self, entry):
+        super().__init__(
+            file_name=basename(entry),
+            source_names=[entry],
+        )
+
+    @property
+    def average_uri(self):
+        return self._get_uri(f'{self._obs_id}.avedb')
+
+    @property
+    def decorrelated_uri(self):
+        return self._get_uri(f'{self._obs_id}.ndatdb')
+
+    @property
+    def has_decorrelated_metadata(self):
+        return self._file_name.endswith('.ndatdb')
+
+    @property
+    def has_undecorrelated_metadata(self):
+        return self._file_name.endswith('.orig')
+
+    @property
+    def has_data(self):
+        return (
+            self._file_name.endswith('.orig')
+            or self._file_name.endswith('.ndatdb')
+            or self._file_name.endswith('.avedb')
+        )
+
+    @property
+    def is_last_to_ingest(self):
+        return self._file_name.endswith('.rlogdb')
+
+    @property
+    def prev(self):
+        result = f'{self._obs_id}_prev.jpg'
+        if self.has_undecorrelated_metadata:
+            result = f'{self._obs_id}_un_prev.jpg'
+        return result
+
+    @property
+    def thumb(self):
+        result = f'{self._obs_id}_prev_256.jpg'
+        if self.has_undecorrelated_metadata:
+            result = f'{self._obs_id}_un_prev_256.jpg'
+        return result
+
+    def is_valid(self):
+        return True
+
+    def rename(self, original_extension, new_extension):
```

Comment:
I think the name is misleading. The calling code could maybe use `replace` to avoid using this method.

Reply:
The content of the method is purely temporal in cohesion, and the point of having the method is doing the two things, one right after the other. I've renamed the method to `use_different_file`, but because there are four locations where it's called, there are four lines of code instead of eight.